index.bs

```diff
@@ -663,6 +663,18 @@ precomposed character).</p>
 <!-- Add some content here about cues and serialisation format in general -->
 <!-- Describe metadata, caption/subtitle, chapter & description cues -->
 
+<p class=note>The box model of WebVTT consists of three key elements: the video viewport, cues, and
+regions. The video viewport is the rendering area into which cues and regions are rendered. Cues are
+boxes consisting of a set of cue lines. Regions are subareas of the video viewport that are used to
+group cues together. Cues are positioned either inside the video viewport directly or inside a
+region, which is positioned inside the video viewport.</p>
+
+<p class=note>A set of cue settings position cues inside the video viewport. A set of region
+settings position regions inside the video viewport. Cues that are inside regions can only use a
+limited set of their cue settings. Specifically, if the cue has a "vertical", "line" or "size"
+setting, the cue drops out of the region. Otherwise, the cue's width is calculated to be relative to
+the region width rather than the viewport. </p>
```

Comment:
I'd like to see an extra sentence in here clarifying if regions act as clipping area for the cues within them or if they are only positioning constructs.

Reply:
"Clipping" is not the right word: if a cue text is longer than the region width allows, it will be broken into another line - that's what CSS does for us.

The definition of region width already states that a cue is to be rendered within the region's width: https://www.w3.org/TR/webvtt1/#webvtt-region-width

> A width
> A number giving the width of the box within which the text of each line of the containing cues is to be rendered, to be interpreted as a percentage of the video width. Defaults to 100.

You'll notice that if the cue has an explicit "size" setting, it drops out of the region. A cue's default size is 100%. For regions this is relative to the region width. Therefore, the cue's width is the same as the region's width.

How would you like me to say this?

Reply:
Gosh, right, there are lots of things to unpack here.

Firstly, in terms of clipping behaviour I think you're saying that there is a combination of wrapping and clipping, so that the clipping only ever occurs in the block progression direction, by means of losing whole lines. The consequence is that no cue text is ever displayed outside the region that the cue is in. However if the text size is such that the whole cue does not fit within the region, there may be some lines of text that are _never visible_. The general rule is that lines that do not fit in their cue's region are not displayed. Those points are worth stating.

Secondly, I can't see in the parsing and processing models why cues with a "size" setting drop out of the region. If they drop out, where do they go? Are they dropped on the floor, or displayed somewhere else outside the region?

The same is true for the "vertical" and "line" settings: in all three cases I could not find algorithmic steps switched on the presence or absence of those settings that modify the position to be either region-based or viewport-based. Please could you point me to where that happens?

Reply:
"Dropping out of a region" merely means that they are not rendered within the region, but rendered in their own right with the settings given.

Looks like there are some things missing in the parsing algorithm. I'll add some text. Thanks for noticing.

Reply:
You're welcome. By the way, this is _exactly_ the kind of thing that's in my mind when I grump about algorithmic specifications and the ability for the WG to test that they do what they're supposed to do.